src/Simulator/IncrementalRedstoneSimulator/RedstoneHandlerFactory.cpp

```diff
@@ -0,0 +1,99 @@
+
+#include "Globals.h"
+#include "RedstoneHandlerFactory.h"
+
+#include "World.h"
+#include "Chunk.h"
+#include "CommandBlockHandler.h"
+#include "DoorHandler.h"
+#include "Blocks/BlockDoor.h"
+#include "RedstoneHandler.h"
+#include "RedstoneTorchHandler.h"
+#include "RedstoneWireHandler.h"
+#include "RedstoneRepeaterHandler.h"
+#include "RedstoneToggleHandler.h"
+#include "SolidBlockHandler.h"
+#include "RedstoneLampHandler.h"
+#include "RedstoneBlockHandler.h"
+#include "PistonHandler.h"
+#include "SmallGateHandler.h"
+#include "NoteBlockHandler.h"
+#include "TNTHandler.h"
+#include "PoweredRailHandler.h"
+#include "PressurePlateHandler.h"
+#include "TripwireHookHandler.h"
+#include "DropSpenserHandler.h"
+#include "RedstoneComparatorHandler.h"
+#include "TrappedChestHandler.h"
+
+
+
+
+
+ComponentPtr cRedstoneHandlerFactory::GetComponent(cWorld & a_World, BLOCKTYPE a_BlockType, cIncrementalRedstoneSimulatorChunkData * a_Data)
+{
+	auto Result = a_Data->m_Handlers.find(a_BlockType);
+	if (Result == a_Data->m_Handlers.end())
+	{
+		auto Handler = cRedstoneHandlerFactory::CreateComponent(a_World, a_BlockType, a_Data);
+		a_Data->m_Handlers[a_BlockType] = Handler;
+		return Handler;
+	}
+	return Result->second;
```

Comment:
No real idea, I just don't think the additional complexity of caching is worthwhile.